### DISCUSSION TITLE
How can I know where I have to plug a laser on a CNC card

### DISCUSSION THREAD

**Discussion by Paugeman09** [vlachoudis/bCNC#1632] [Q&A]
Here is my material

**Comment by Paugeman09** [comment]
![image](https://user-images.githubusercontent.com/44941844/119733439-cfb50780-be79-11eb-8e61-f326c4105518.jpeg)
![image](https://user-images.githubusercontent.com/44941844/119733441-cfb50780-be79-11eb-85ea-6edea75c0f0a.jpeg)
![image](https://user-images.githubusercontent.com/44941844/119733442-d04d9e00-be79-11eb-9d0b-1792b9618199.jpeg)

**Comment by carl1961** [comment]
Most connect to spindle pin which is usually pin D12 
Read up on your laser module, it may need a driver module, 

check your    cpu_map_h


https://craftweeks.com/en/how-to-steup-grbl-for-laser-engravers/

**Comment by rjd1234** [comment]
I typically generate toolpaths in Fusion 360 and the Laser cutting post processors that I have used activate the laser using S1000 and then M3 and M5... This is the same way that a spindle would be turned on it off. 

So, yes. Connect it to the spindle terminals.

**Comment by Paugeman09** [comment]
> I typically generate toolpaths in Fusion 360 and the Laser cutting post processors that I have used activate the laser using S1000 and then M3 and M5... This is the same way that a spindle would be turned on it off. 
> 
> 
> 
> So, yes. Connect it to the spindle terminals.

What do you call a spindle terminal ?

**Comment by Paugeman09** [comment]
> Most connect to spindle pin which is usually pin D12 
> 
> Read up on your laser module, it may need a driver module, 
> 
> 
> 
> check your    cpu_map_h
> 
> 
> 
> 
> 
> https://craftweeks.com/en/how-to-steup-grbl-for-laser-engravers/

So if I understood I just have to plug it in the pin D12, but my laser have three ports, which one should I plug ?
What do you call a driver Modul ? 
Thank you for your help

**Comment by Harvie** [comment]
Two pins are for power, you need to connect to suitable power supply according to laser specifications.
Third pin is on/off PWM signal that can be provided by your controller board directly.

**Comment by Paugeman09** [comment]
> Two pins are for power, you need to connect to suitable power supply according to laser specifications.
> 
> Third pin is on/off PWM signal that can be provided by your controller board directly.

Thank you very much for your help, I just have a last question, how can I know which pin is the power pin ?

**Comment by Harvie** [comment]
Why don't you ask manufacturer/distributor of that laser?
I would expect black/red/yellow to be gnd/12V/signal. BUT IF I AM WRONG, YOU WILL DAMAGE THE LASER PERMANENTLY.
So ask your vendor for manual/datasheet.

**Comment by Harvie** [comment]
Also this laser is powerfull enough to permanently damage your vision just by short glimpse of reflection from little piece of reflective surface. You need at least wavelength specific glasses, but more likely fully enclosed machine. Also can set your house on fire if left unsupervised. And you need proper ventilation ducts to suck the smoke from cutting, fumes can be toxic.

**Comment by Paugeman09** [comment]
> Why don't you ask manufacturer/distributor of that laser?
> 
> I would expect black/red/yellow to be gnd/12V/signal. BUT IF I AM WRONG, YOU WILL DAMAGE THE LASER PERMANENTLY.
> 
> So ask your vendor for manual/datasheet.

I will try to contact him thank you I hope he respond

**Comment by Paugeman09** [comment]
> Also this laser is powerfull enough to permanently damage your vision just by short glimpse of reflection from little piece of reflective surface. You need at least wavelength specific glasses, but more likely fully enclosed machine. Also can set your house on fire if left unsupervised. And you need proper ventilation ducts to suck the smoke from cutting, fumes can be toxic.

Thank you for the prevention I had buy some protection glasses
